src/python/pants/backend/python/goals/run_pex_binary.py

```diff
@@ -114,12 +123,61 @@ def in_chroot(relpath: str) -> str:
     ]
     extra_env = {
         **complete_pex_env.environment_dict(python_configured=pex.python is not None),
-        "PEX_PATH": in_chroot(local_dists.pex.name),
+        "PEX_PATH": _in_chroot(local_dists.pex.name),
         "PEX_EXTRA_SYS_PATH": os.pathsep.join(source_roots),
     }
 
     return RunRequest(digest=merged_digest, args=args, extra_env=extra_env)
 
 
+@rule
+async def run_pex_debug_adapter_binary(
+    field_set: PexBinaryFieldSet,
+    debugpy: DebugPy,
+    debug_adapter: DebugAdapterSubsystem,
+) -> RunDebugAdapterRequest:
+    if field_set.run_in_sandbox:
+        logger.warning(
+            softwrap(
+                """
+                Using --debug-adapter with `run_in_sandbox` set to `True` will likely cause your
```

Comment:
```suggestion
                Using `--run-debug-adapter` on the target {field_sett.address}, which sets the field
                `run_in_sandbox` to `True`. This will likely cause your ...
```